app/scripts/metamask-controller.js

```diff
@@ -1374,9 +1379,15 @@ export default class MetamaskController extends EventEmitter {
     networkControllerMessenger.subscribe(
       'NetworkController:networkDidChange',
       async () => {
-        const { ticker } = this.networkController.state.providerConfig;
         try {
-          await this.currencyRateController.setNativeCurrency(ticker);
+          if (
+            this.preferencesController.store.getState().useCurrencyRateCheck
+          ) {
+            await this.currencyRateController.stopAllPolling();
+            this.currencyRateController.startPollingByNetworkClientId(
+              this.networkController.state.selectedNetworkClientId,
+            );
+          }
```

Comment:
Is this changing the behavior here? Previously calling `setNativeCurrency` didn't start polling it just [updated the exchangerate once](https://github.com/MetaMask/core/pull/1805/files#diff-9c8bbd91f0dff6d7000e65707a7045d650c9eaaa82984c2ff911ef559870a19bL181)?

Reply:
not sure why we wouldn't want to poll here but I'm not seeing that we did before?

Reply:
It does not as we guard starting the poll with a call to `this.preferencesController.store.getState().useCurrencyRateCheck`

Reply:
right but previously we didn't start polling regardless of whether or not the `useCurrencyRateCheck` was set to true?

Reply:
Right, previously we did not start/stop polling for the CurrencyRateController on networkDidChange from MetaMaskController. That's because previously, it was possible to tell the CurrencyRateController that you wanted to "queue up" an update for the exchange rates for XYZ native currency and let the controller poll for it on its own time whenever that next poll iteration would happen. Now it's not possible to tell the CurrencyRateController to "queue up" a nativeCurrency. It's either polling or it isn't. Because of that, we need to keep track of `useCurrencyRateCheck` outside of the CurrencyRateController and handling the starting/stopping of polling ourselves

Reply:
ah ok gotcha. Last question: why not have the CurrencyRateController's messenger subscribe to NetworkController changes and do this start/stop polling in the controller itself as we do with most other controllers? I see that we have to do this pattern for subscribing to other controllers state which haven't migrated to the controllers v2 pattern yet but seems we could in this case.

Reply:
> why not have the CurrencyRateController's messenger subscribe to NetworkController changes and do this start/stop polling in the controller itself as we do with most other controllers?

I suppose it could. That would require adding a guard to check `useCurrencyRateCheck` on each polling loop as a way to pause polling as opposed to just stopping the polling itself directly.  That would mean if `useCurrencyRateCheck` is set to false, and i call `currencyRateController.startPollingByNetworkClientId('123')`, then the polling would be active, but will always exit early until `useCurrencyRateCheck` is set to true.

Does this seem to be more inline with how you were thinking we should approach this?

Reply:
Could we not check `useCurrencyRateCheck` in the frontend an only call `startPollingByNetworkClientId` if its true?

Reply:
Oh sorry that didn't make sense in this context.

Reply:
Well couldn't we check `useCurrencyRateCheck` in the handler for `networkStateChange` on the `CurrencyRateController` and only start polling if its true?

Reply:
I won't block on this. We can leave it as is for now.